packages/react-reconciler/src/ReactFiber.js

```diff
@@ -277,6 +280,8 @@ function FiberNode(
   this.sibling = null;
   this.index = 0;
 
+  this.residue = null;
```

Comment:
based on discussion with @sebmarkbage will remove residue and replace with a childFiber-like comparison algorithm that uses identical semantics as the reconciliation step but does not actually reconcile.

This is so we can determine if we are returning identical children from a previous render and can avoid reifying the speculative work tree